bootcamp-milestone-3A/src/app/blog/[slug]/route.ts

```diff
@@ -0,0 +1,37 @@
+import { NextRequest, NextResponse } from "next/server";
+import connectDB from "../../database/db";
+import blogSchema from "../../database/blogSchema";
+
+type IParams = {
+  params: {
+    slug: string;
+  };
+};
+
+/* 
+	In order to use params, you need to have a request parameter before
+
+	The reason why we do { params }, is to destructure, the object, meaning,
+	it allows us to obtain the individual properties within the "IParams" 
+	object directly and conveniently, 
+	such as the `params` property.
+
+	If we didn't do this, to obtain slug would look messy,
+	ex.
+	const slug = params.params.slug
+
+	There are more ways to destructure this, but that is up to you to find out
+	lol.
+
+ */
+export async function GET(req: NextRequest, { params }: IParams) {
```

Comment:
This file should go into `app/api` folder. In 3B, this will hold your frontend code for a dynamic blog page instead.